core/shared/src/main/scala/zio/internal/Hub.scala

```diff
@@ -76,7 +76,8 @@ private[zio] object Hub {
    * Constructs a new bounded hub with the requested capacity.
    */
   def bounded[A](requestedCapacity: Int): Hub[A] = {
-    assert(requestedCapacity > 0)
+    // NOTE: Do not use `assert` as the compiler removes it in releases
+    require(requestedCapacity > 0)
```

Comment:
We need to use `require` for testing user provided arguments so that they're not removed in release artifacts. This is also documented in the scaladoc of `require`